designs/2022-unused-disable-directive-flexible-config/README.md

```diff
@@ -0,0 +1,225 @@
+
+We will also change the default behavior to warn on unused disable directives so that all users (unless they opt-out) will benefit from the detection (and automatic removal with `--fix`) of unused disable directives.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+How the new "explicit severity" design works:
+
+- A new CLI option `--report-unused-disable-directives-severity <severity>` is added, while the existing CLI option `--report-unused-disable-directives` remains in place untouched
+- If both CLI options `--report-unused-disable-directives` and `--report-unused-disable-directives-severity <severity>` are provided, an error is thrown
+- The existing config file option `reportUnusedDisableDirectives` is updated to, in addition to the current boolean values, also accept standard severity level string values: `off`, `warn`, `error` (or the corresponding number for each level)
+- If both the CLI and config file options are present, the CLI option takes precedence over (overrides) the config file option
+- If only the CLI option is present, then it is used; if only the config file option is present, then it is used
+- If neither the CLI nor config file options are present, the default value (`warn`) is used
+- Note: There's only one underlying setting, and the config and CLI options are just two different ways of controlling it
+
+Allowed values for the config file option:
+
+- `reportUnusedDisableDirectives: true` - same behavior as today (`warn`)
+- `reportUnusedDisableDirectives: false` - same behavior as today (`off`)
+- `reportUnusedDisableDirectives: "error"` - new value (or `2`)
+- `reportUnusedDisableDirectives: "warn"` - new value (or `1`)
+- `reportUnusedDisableDirectives: "off"` - new value (or `0`)
+
+Allowed values for the CLI options:
+
+- `--report-unused-disable-directives` - existing option, same behavior as today (`error`)
+- `--report-unused-disable-directives-severity error` - new option (or `2`)
+- `--report-unused-disable-directives-severity warn` - new option (or `1`)
+- `--report-unused-disable-directives-severity off` - new option (or `0`)
+
+### Phases
+
+The implementation of this RFC will likely involve two phases:
+
+1. Phase 1: A non-breaking change to add support for the new severity levels.
+2. Phase 2: A breaking change intended for a major release in which the default behavior is changed to `warn`.
```

Comment:
Since the default behavior will be changed only for flat config users, perhaps we could change it in a minor release? @nzakas what do you think?

Reply:
I'm open to that. I made some clarifications for phase 2:

> Phase 2: Any breaking changes for flat config users (`warn` by default, remove redundant `reportUnusedDisableDirectives` option from `processOptions()`), suitable for a major release or potentially a minor release since the flat config is not considered stable yet.

Let me know if phase 2 should fall in minor or major release.

Reply:
We can update the default behavior for flat config prior to v9 in a minor release.

If we change it after v9, then it would need to be in a major release.

Reply:
Sounds good. I have updated the phase 2 wording to account for this. I believe I've addressed all feedback on the RFC.